subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -5407,7 +5407,9 @@ static inline int ctrl_rx(memq_link_t *link, struct node_rx_pdu **rx,
 #endif /* CONFIG_BT_CTLR_SLAVE_FEAT_REQ */
 
 	case PDU_DATA_LLCTRL_TYPE_FEATURE_RSP:
-		if (!pdu_len_cmp(PDU_DATA_LLCTRL_TYPE_FEATURE_RSP,
+		if ((!IS_ENABLED(CONFIG_BT_CTLR_SLAVE_FEAT_REQ) &&
```

Comment:
Conditional compile the complete block and other associated code (`feature_rsp_recv`) with:
`#if defined(CONFIG_BT_CTLR_SLAVE_FEAT_REQ) || defined(CONFIG_BT_CENTRAL)`

Reply:
That would not work in a configuration with `CONFIG_BT_CENTRAL=y`, `CONFIG_BT_PERIPHERAL=y`, and `CONFIG_BT_CTLR_SLAVE_FEAT_REQ=n` (i.e., where the same device can be master and slave).

Reply:
if `CONFIG_BT_CTLR_SLAVE_FEAT_REQ=n` then only central role is to receive the `PDU_DATA_LLCTRL_TYPE_FEATURE_RSP`, and the `default` case will generate the unknown response in slave role.

Reply:
I stand corrected. The original changes look good (where the same device can be master and slave, but not support slave feature req).